app/src/main/java/fi/helsinki/opintoni/service/FeedbackService.java

```diff
@@ -17,42 +17,95 @@
+    private static final DateTimeFormatter TIMESTAMP_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    private static final String DEFAULT_LANG = Language.FI.getCode();
+    private static final String COUNTRY_FINLAND_CODE = "FI";
+    private static final String FIELD_TIMESTAMP = "timestamp";
+    private static final String FIELD_FACULTY = "faculty";
+    private static final String FIELD_STATE = "state";
+    private static final String FIELD_USER_AGENT = "userAgent";
+    private static final String FIELD_LANG = "lang";
+    private static final String DEFAULT_SUBJECT = "Palaute";
 
+    private final MailSender mailSender;
+    private final MessageSource messageSource;
     private final FeedbackRepository feedbackRepository;
-    private final ObjectMapper objectMapper;
     private final FeedbackConverter feedbackConverter;
+    private final String efecteEmailAddress;
+    private final String anonymousFeedbackFromAddress;
+    private final String anonymousFeedbackReplyToAddress;
+
 
     @Autowired
-    public FeedbackService(FeedbackRepository feedbackRepository,
-                           ObjectMapper objectMapper,
+    public FeedbackService(MailSender mailSender,
+                           MessageSource messageSource,
+                           @Value("${feedback.efecte.address}") String efecteEmailAddress,
+                           @Value("${feedback.anonymous.fromAddress}") String anonymousFeedbackFromAddress,
+                           @Value("${feedback.anonymous.replyToAddress}") String anonymousFeedbackReplyToAddress,
+                           FeedbackRepository feedbackRepository,
                            FeedbackConverter feedbackConverter) {
+        this.mailSender = mailSender;
+        this.messageSource = messageSource;
+        this.efecteEmailAddress = efecteEmailAddress;
+        this.anonymousFeedbackFromAddress = anonymousFeedbackFromAddress;
+        this.anonymousFeedbackReplyToAddress = anonymousFeedbackReplyToAddress;
         this.feedbackRepository = feedbackRepository;
-        this.objectMapper = objectMapper;
         this.feedbackConverter = feedbackConverter;
     }
 
-    public void insertFeedback(InsertFeedbackRequest request) throws Exception {
-        Feedback feedback = new Feedback();
-        feedback.content = request.content;
-        feedback.email = request.email;
-        feedback.metadata = objectMapper.writeValueAsString(request.metadata);
-        feedbackRepository.save(feedback);
+    public void insertFeedback(InsertFeedbackRequest request) throws MailException {
+        String content = request.content;
+        String timestamp = createTimestamp();
+        String state = getState(request);
+        String faculty = getFaculty(request);
+        String userAgent = getUserAgent(request);
+        Locale locale = getLocale(request);
+        String body = buildMessageBody(content, timestamp, faculty, state, userAgent, locale);
```

Comment:
`buildMessageBody` voisi saada parametrina `request`in, jolloin rivit 91-96 voisi siirtää `buildMessageBody`yn mihin ne semanttisemmin kuuluvat (ainoastaan `buildMessageBody` tarvitsee tietää mitä bodyn sisällöksi generoidaan).